### DISCUSSION TITLE
how to perform netmiko.send_command_timing() method on napalm?

### DISCUSSION THREAD

**Discussion by teliahuseyinturhan** [napalm-automation/napalm#1908] [Q&A] [answered]
Hi everybody, I'm a beginner on network automation. Basically, I am trying to use napalm library to create a ethernet loopback on a cisco device. But I have an issue at some point after editing the interface, which is running this command
 `device.cli(['ethernet loopback start .... ', 'yes'])`.
It gives a ReadTimeout error, because, I think it can't escape from ios_cli. The suggestion to fix is:
```
1. Explicitly set your pattern using the expect_string argument.
2. Increase the read_timeout to a larger value. You can also look at the Netmiko session_log or debug log for more information.
```
But napalm.cli() method doesn't accept expect_string argument.
I can perform same task using netmiko library with this set of commands 
```
net_connect.send_command_timing('ethernet loopback start ....')
net_connect.send_command_timing('yes')
```
Netmiko.send_command_timing() method helps me as it requires a confirmation for the device to proceed `This is an intrusive loopback and the packets matched with the service will not  be able to pass through. Continue? (yes/[no]):`. 

How can I accomplish same task (steps) using napalm.cli() method? Thanks in advance.

**Comment by ktbyers** [comment, marked as answer]
.cli() doesn't support that.

I would just do the following:

```python
# napalm_obj.device is the Netmiko connection object
netmiko_conn = napalm_obj.device
netmiko_conn.send_command_timing(...)

# or, alternatively
netmiko_conn.send_command(cmd, expect_string="whatever")
```

**Reply by teliahuseyinturhan** [reply]
Thanks for response. As I mentioned in the question I have no issue to perform this task using netmiko library using either option you recommended.
And I'm convinced that I cannot accomplish the task using Napalm, so I will not be spending more time on that. Thanks again for the support.

**Reply by ktbyers** [reply]
@teliahuseyinturhan Above, I am showing you how to use it via NAPALM i.e. NAPALM drivers that use Netmiko give you direct access to the Netmiko connection object.

**Comment by teliahuseyinturhan** [comment]
It perfectly works in both way as `netmiko_conn = napalm_obj.device` or `netmiko_conn = napalm_obj._netmiko_device` even though it shows this warning before running the script 
`"send_command_timing" is not a known member of 
"None" Pylance [reportOptionalMemberAccess]`. 

But as I said it solved my issue. Thanks again for all your support.
